submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,358 @@
+
+const maxNumberOfFriends = 100;
+const randomUserUrl = 'https://randomuser.me/api/?results=' + maxNumberOfFriends + '&seed=friends';
+
+const minPersonAge = 18;
+const maxPersonAge = 99;
+
+const state = {
+  friends: [],
+  filters: {
+    partOfName: '',
+    gender: '',
+    genderList: [],
+    ageRange: [minPersonAge, maxPersonAge],
+    country: '',
+    countryList: [],
+  },
+  sorter: {
+    keyName: '',
+    order: '',
+    fieldPathList: {
+      name: ['name', 'first'],
+      age: ['dob', 'age'],
+      country: ['location', 'country'],
+    },
+    orderSymbols: {
+      ASC: '&#9650;', 
+      DESC: '&#9660;',
+    },
+  },
+  initialListLength: 25,
+  numberOfShowedFriends: 0,
+  scrollDisabled: false,
+  afterLoadingMoreFriendsScrollShift: 25, /* px */
+  nextMoreFriendAutoloadDelay: 300,       /* ms */
+}
+
+const preloader = document.querySelector('.preloader');
+const container = document.querySelector('.container');
+
+const sortField = document.querySelector('.sort-field');
+const sortOrder = document.querySelector('.sort-order');
+const resetButton = document.querySelector('.reset');
+
+const searchField = document.querySelector('.search-field');
+const filterGender = document.querySelector('.filter-gender');
+const filterMinAge = document.querySelector('.filter-min-age');
+const filterMaxAge = document.querySelector('.filter-max-age');
+const filterCountry = document.querySelector('.filter-country');
+
+const drawPerson = ({location, phone, email, gender, name, picture, dob}) => {  
+  const {street, city, state, country} = location;
+  const div = document.createElement('div');
```

Comment:
you can create this `div` with template strings as well. it will help you to decrease DOM operations. And just use `container.innerHTML` for inserting string content into the `container`

Reply:
Very good comment! 
I have to clarify my solution.
Unfortunately, assignment to innerHTML causes the destruction of all child elements, even if you're trying to append.
In this case, every new card in the container makes all cards blink for a moment. And the time of working of each personDraw() increase in 5 times. 
You can see this difference on these screenshots:
[Timeline of drawPerson() with DOM manipulation](https://raw.githubusercontent.com/andrewklmn/kottans-frontend/main/friends_app/drawPerson_timeline_with_dom_operations.png)
[Timeline of drawPerson() without DOM manipulation](https://raw.githubusercontent.com/andrewklmn/kottans-frontend/main/friends_app/drawPersone_timeline_without_dom_manipulating.png)

So I did it in this way just for improving animation performance.

Reply:
ok! nice argument! appreciate it